python/src/scipp/_cumulative.py

```diff
@@ -1,11 +1,16 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
 # @author Simon Heybrock
+
+from typing import Optional
+
 from ._scipp import core as _cpp
 from ._cpp_wrapper_util import call_func as _call_cpp_func
 
 
-def cumsum(a, dim=None, mode='inclusive'):
+def cumsum(a: _cpp.Variable,
+           dim: Optional[str] = None,
+           mode: Optional[str] = 'inclusive') -> _cpp.Variable:
```

Comment:
True. This affects basically all functions which aren't supported with Dataset yet. We could relax this and just require any of `Variable`, `DataArray`, and `Dataset`. This would be better than nothing concerning type checkers but it would be confusing to people looking at the docs and getting type errors as a result.